src/main/java/controller/CarController.java

```diff
@@ -1,55 +1,72 @@
+import utils.StringUtils;
 import view.InputView;
 import view.OutputView;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
 
 public class CarController {
 
-    private final InputView inputView;
-    private final OutputView outputView;
-    private final CarService carService;
-
-    public CarController(InputView inputView, OutputView outputView, CarService carService) {
-        this.inputView = inputView;
-        this.outputView = outputView;
-        this.carService = carService;
-    }
+    private Cars cars;
 
     public void run(RacingNumberGenerator generator) {
-        generateCars();
-        Round round = generateRound();
+        try {
+            cars = generateCars(InputView.inputCarsName());
+            Round round = generateRound();
 
-        race(generator, round);
-        outputView.printWinner(carService.getWinner());
+            race(generator, round);
+            OutputView.printWinner(cars.getWinner());
+        } catch (IllegalArgumentException exception) {
+            OutputView.printExceptionMessage(exception.getMessage());
+            run(generator);
+        }
     }
 
     private void race(RacingNumberGenerator generator, Round round) {
-        outputView.printRoundStartMessage();
+        OutputView.printRoundStartMessage();
         for (int count = 0; count < round.getRound(); count++) {
-            carService.race(generator);
-            outputView.printRound(carService.getCarsDto());
+            cars.race(generator);
+            OutputView.printRound(cars.getCarsDto());
         }
     }
 
-    private void generateCars() {
+    private Round generateRound() {
         try {
-            String carsName = inputView.inputCarsName();
-            carService.generateCars(carsName);
+            return new Round(InputView.inputRound());
         } catch (IllegalArgumentException exception) {
-            outputView.printExceptionMessage(exception.getMessage());
-            generateCars();
+            OutputView.printExceptionMessage(exception.getMessage());
+            return generateRound();
         }
     }
 
-    private Round generateRound() {
-        try {
-            return new Round(inputView.inputRound());
-        } catch (IllegalArgumentException exception) {
-            outputView.printExceptionMessage(exception.getMessage());
-            return generateRound();
+    public Cars generateCars(String inputCarsName) {
+        String[] carsName = StringUtils.splitBySeparator(inputCarsName);
+        checkDuplication(carsName);
+        return new Cars(mapToCars(carsName));
+    }
+
+    private List<Car> mapToCars(String[] carsName) {
+        return Arrays.stream(carsName)
+                .map(Car::new)
+                .collect(Collectors.toList());
+    }
+
+    private void checkDuplication(String[] carsName) {
+        if (getDistinctCarsCount(carsName) != carsName.length) {
+            throw new DuplicateCarNameException();
         }
```

Comment:
```suggestion
        checkDuplication(carsName);
        return new Cars(mapToCars(carsName));
    }

    private void checkDuplication(String[] carsName) {
        if (getDistinctCarsCount(carsName) != carsName.length) {
            throw new DuplicateCarNameException();
    }

    private List<Car> mapToCars(String[] carsName) {
        return Arrays.stream(carsName)
                .map(Car::new)
                .collect(Collectors.toList());
    }
```

사용되는 순서에 따라 메서드를 정렬해주면 코드 가독성이 더 올라갈 거예요 !

Reply:
감사합니당!